services/thermal_mgr/thermal_mgr.c

```diff
@@ -2,70 +2,112 @@
 
+
 static void thermalMgr(void *pvParameters);
 
+
 void initThermalSystemManager(lm75bd_config_t *config) {
-  memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
-  memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+ memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
+ memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+   thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+   config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+
+
+ memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
+ memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
+
 
-  memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
-  memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
+ thermalMgrQueueHandle = xQueueCreateStatic(
+   THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+   thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 
-  thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+ /* Send an event to the thermal manager queue */
+ if(event==NULL)return ERR_CODE_INVALID_ARG;        //if invalid argument
+ if(thermalMgrQueueHandle==NULL)return ERR_CODE_INVALID_STATE;      //fail if thermalmgrqueuehandle doesnt work
+ if(xQueueSend(thermalMgrQueueHandle,event,0)==pdPASS) return ERR_CODE_QUEUE_FULL;   //if queue is full
+ return ERR_CODE_SUCCESS;
 }
 
+
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+ /* Implement this function */
+ thermal_mgr_event_t event = {.type=THERMAL_MGR_EVENT_OVER_TEMP_RECEIVED};
+ thermalMgrSendEvent(&event);
 }
 
+
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t event;
+    float tempval;
+    if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE) {
+      // Only read the temperature when the event type is THERMAL_MGR_EVENT_MEASURE_TEMP_CMD
+      if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
+        readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempval);
+        addTemperatureTelemetry(tempval);
+      } 
+      else if (event.type == THERMAL_MGR_EVENT_OVER_TEMP_RECEIVED) {
+        readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempval);
+        if (tempval >= LM75BD_DEFAULT_OT_THRESH) {
+          overTemperatureDetected();
+        }
+        if (tempval <= LM75BD_DEFAULT_HYST_THRESH) {
+          safeOperatingConditions();
+        }
+      }
+      else{
+        LOG_ERROR("Invalid event received");
```

Comment:
Better to define an error code that represents an invalid event than logging a string